tests/Unit/Evolution/Systems/NewtonianEuler/Test_Characteristics.cpp

```diff
@@ -321,6 +321,18 @@ void test_right_and_left_eigenvectors() noexcept {
   const EquationsOfState::PolytropicFluid<false> polytropic_eos{100., 2.};
   test_left_and_right_eigenvectors_impl(
       density, velocity, specific_internal_energy, unit_normal, polytropic_eos);
+
+  // Test also in cases where unit normal is aligned to coordinate axes
+  for (size_t i = 0; i < Dim; ++i) {
```

Comment:
```
for (const auto& direction : Direction<Dim>::all_directions()) {
  const auto aligned = euclidean_basis_vector(direction, used_for_size);
  test_left_and_right_eigenvectors_impl(
      density, velocity, specific_internal_energy, aligned, ideal_gas_eos);

  test_left_and_right_eigenvectors_impl(
      density, velocity, specific_internal_energy, aligned, polytropic_eos);
}
```

should give the test along all directions for all dimensions for the same number of lines. However, I noticed that `euclidean_basis_vector` only works for `DataVectors`, for which I pushed 4e473ff2b63775ebdfa605bb454403e015d9f661 in #1958 as I also use it for relativistic hydro. Feel free to cherry-pick that commit and update your PR, or leave it as it is.